packages/framework-core/src/booster-command-dispatcher.ts

```diff
@@ -41,10 +42,17 @@ export class BoosterCommandDispatcher {
 
     const commandInstance = createInstance(commandClass, commandInput)
 
-    const register = new Register(commandEnvelope.requestID, commandEnvelope.currentUser, commandEnvelope.context)
+    const register = new Register(
+      commandEnvelope.requestID,
+      commandEnvelope.responseHeaders,
+      commandEnvelope.currentUser,
+      commandEnvelope.context
+    )
+
     this.logger.debug('Calling "handle" method on command: ', commandClass)
     const result = await commandClass.handle(commandInstance, register)
     this.logger.debug('Command dispatched with register: ', register)
+    context.responseHeaders = register.responseHeaders
```

Comment:
An advantage of setting in `register.responseHeaders` a reference to `context.responseHeaders` is that we also avoid the need of re-setting it here.

```suggestion
```